swri_yaml_util/CMakeLists.txt

```diff
@@ -32,10 +35,14 @@ include_directories(SYSTEM
   ${YamlCpp_INCLUDE_DIR}
 )
 
+if ($ENV{ROS_DISTRO} STREQUAL "kinetic")
```

Comment:
I dont think we should be setting the global CMAKE_CXX standard here, or even changing it for kinetic at all in this case.

Reply:
You're right that now with the ifdef on the pointer type, I don't think there are any >=c++11 features for this library. I was originally trying to be consistent with the standard setting across the whole marti_common "stack" so that a kinetic user that wants to link to multiple marti_common libraries does not have to worry about ABI issues.

I will go ahead and set it per target, because I still think the whole stack should be >=c++11 compatible.

Reply:
Well, the issue is we aren't really just making it C++11 compatible, we're potentially breaking backwards compatibility as well. It's probably fine, but it would be a change from previous.

Reply:
I dove down the rabbit hole a bit to try and find out why we weren't alread having ABI problems with existing c++11 applications that link with swri_yaml_util (assuming they had binaries or built it from source using the c++03 standard flags). It seems that the answer is [dual ABI](https://gcc.gnu.org/onlinedocs/libstdc++/manual/using_dual_abi.html). xenial / kinetic is a somewhat special release in that *most* users are getting binaries built with gcc >= 5.2 and the macro `_GLIBCXX_USE_CXX11_ABI=1` as the default setting. The use of `std::string` throughout the swri_yaml_util and yaml-cpp library APIs worried me a bit since that was one of the major ABI breaking changes from c++03 to c++11, but I think this is exactly where these default settings are saving us.

I've experimented a bit with linking swri_yaml_util with an executable that uses a couple of the APIs that require `std::string` and set every permutation of the standard properties, turned on `-Wabi`, and manually set `-D_GLIBCXX_USE_CXX11_ABI=0`. Manually setting the macro was the only way I could get the linker to fail. All of that to say: for most kinetic users, it probably doesn't matter.

Kind of a moot point I think because kinetic is no longer receiving updates and the users that would apply this patch in kinetic would likely build from source, it wouldn't get rid of the `std::auto_ptr` warning for them, and hopefully they'll be moving on to a more modern distribution soon:)

I will remove the standard spec properties from this package to reduce the chance of something breaking if someone is going to build with a super old or different compiler toolchain.